app/models/rbac.rb

```diff
@@ -248,7 +248,33 @@ def self.find_targets_with_user_group_rbac(klass, scope, rbac_filters, find_opti
     end
   end
 
+  def self.group(user_or_group)
+    case user_or_group
+    when User
+      user_or_group.current_group
+    when MiqGroup
+      user_or_group
+    when NilClass
```

Comment:
Great, thanks... yeah, I threw this in here to avoid putting in the actual search method knowing it didn't belong here.  This would be where a method such as `visible_tenants` on the user, group, etc. could give us the knowledge of walking the tenant ancestry to give the right tenants we should be able to see within.  What exactly is the ancestry visibility of the following cases:
```
        a (default tenant)
      / |   
   aa  ab
  /     |
aaa    abb

c  (no tenant)
```

For each tenant, who can see what?  I can put some test cases around it and see about how to implement them.